packages/notification/channel/instatus/database.ts

```diff
@@ -54,9 +59,25 @@ export async function findIncident({
   url,
 }: FindIncident): Promise<FindIncidentResponse | undefined> {
   const sqlStatement = `SELECT incident_id FROM instatus_page_incidents 
-    WHERE status = ? AND url = ? AND probe_id = ?`
+    WHERE status = ? AND url = ? AND probe_id = ? LIMIT 1`
   const sqlParams = [status, url, probeID]
-  const incident = await db.get<FindIncidentResponse>(sqlStatement, sqlParams)
+  const incident = await (
+    await getDatabase()
+  ).get<FindIncidentResponse>(sqlStatement, sqlParams)
 
   return incident
 }
+
+async function getDatabase(): Promise<
+  Database<sqlite3.Database, sqlite3.Statement>
+> {
+  if (!db) {
+    db = await open({
+      filename: dbPath,
+      mode: sqlite3.OPEN_READWRITE | sqlite3.OPEN_CREATE,
+      driver: sqlite3.Database,
+    })
+  }
+
+  return db
+}
```

Comment:
should this be here? or should it be in like common directory?

Reply:
IMO, only Instatus will use this code for now. So it would be better if we just put all the Instatus related code in one directory. If in the future other channel need it, the developer can extract the functionality to make it reusable